src/Service/ActivityPubManager.php

```diff
@@ -130,11 +130,13 @@ public function findActorOrCreate(?string $actorUrlOrHandle): null|User|Magazine
         if (false === filter_var($actorUrl, FILTER_VALIDATE_URL)) {
             if (!substr_count(ltrim($actorUrl, '@'), '@')) {
                 $user = $this->userRepository->findOneBy(['username' => ltrim($actorUrl, '@')]);
-                if ($user->apFetchedAt->modify('+1 hour') < (new \DateTime())) {
-                    $this->bus->dispatch(new UpdateActorMessage($user->apProfileId));
-                }
+                if ($user instanceof User) {
+                    if ($user->apFetchedAt->modify('+1 hour') < (new \DateTime())) {
+                        $this->bus->dispatch(new UpdateActorMessage($user->apProfileId));
+                    }
 
-                return $user;
+                    return $user;
+                }
```

Comment:
Should we do a return null? Should we do a `createUser`?

Reply:
If you found the user object you should return the user, like you did

Reply:
And if you mean "should we create the user in the else or return null" my answer would be neither. If the actor doesn't exist it already is created in this method, because we verified that it is a user handle and if we can't find the user locally we do a webfinger request to get the actual URL which we then fetch

Reply:
OK. So this might still return null. Then I think a simple check I added should then be sufficient.